> That error is on Travis not on your system. It doesn't matter whether or not numpy is installed on your computer.

But in [this](https://github.com/sympy/sympy/pull/18192#issuecomment-576734546) PR, I just updated my pip version, closed the issue, reopened it and then the build was successful after that. 
...........................................................................
> The mistake in this PR right now is that it checks for the presence of `__index__` but then calls `int`. 

That's bc I need to deal with the condition when `strict` is False....  
Consider this example:
```
>>> big = 1e23
>>> type(big) is float
True
>>> big == int(big)
True
>>> as_int(big)
Traceback (most recent call last):
 ...
 ValueError: ... is not an integer
 >>> as_int(big, strict=False)
 99999999999999991611392
```
...................................................................................

> It should just be
> 
> ```
> from operator import index
> ...
>     if strict:
>         try:
>             return index(n)
>         except TypeError:
>             raise ValueError(...)
> ```
What about the case when `strict` is False?? I guess you need to call `int` in the `else` block. The latest commit I've added passes all the tests.  Please correct me if I'm wrong, @oscarbenjamin @asmeurer.

--- a/sympy/core/compatibility.py
+++ b/sympy/core/compatibility.py
@@ -350,10 +350,10 @@ def as_int(n, strict=True):
     """
     Convert the argument to a builtin integer.
 
-    The return value is guaranteed to be equal to the input. ValueError
-    is raised if the input has a non-integral value. When ``strict`` is
-    False, non-integer input that compares equal to the integer value
-    will not raise an error.
+    The return value is guaranteed to be equal to the input. ValueError is
+    raised if the input has a non-integral value. When ``strict`` is True, this
+    uses `__index__ <https://docs.python.org/3/reference/datamodel.html#object.__index__>`_
+    and when it is False it uses ``int``.
 
 
     Examples
@@ -396,16 +396,19 @@ def as_int(n, strict=True):
     ...
     ValueError: ... is not an integer
     """
-    from sympy.core.numbers import Integer
-    try:
-        if strict and not isinstance(n, SYMPY_INTS + (Integer,)):
-            raise TypeError
-        result = int(n)
-        if result != n:
-            raise TypeError
+    if strict:
+        try:
+            return operator.index(n)
+        except TypeError:
+            raise ValueError('%s is not an integer' % (n,))
+    else:
+        try:
+            result = int(n)
+        except TypeError:
+            raise ValueError('%s is not an integer' % (n,))
+        if n != result:
+            raise ValueError('%s is not an integer' % (n,))
         return result
-    except TypeError:
-        raise ValueError('%s is not an integer' % (n,))
 
 
 def default_sort_key(item, order=None):

--- a/sympy/core/tests/test_compatibility.py
+++ b/sympy/core/tests/test_compatibility.py
@@ -75,10 +75,3 @@ def test_ordered():
     assert list(ordered(l, warn=True)) == [[1], [1], [2]]
     raises(ValueError, lambda: list(ordered(['a', 'ab'], keys=[lambda x: x[0]],
         default=False, warn=True)))
-
-
-def test_round():
-    from sympy.core.compatibility import round
-    # XXX delete this test when PY2 is dropped
-    assert round(S(12.3)).is_Integer
-    assert round(S(12.3), 0).is_Float

